.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeProcessing.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Recognizers.Text.Utilities;
+using DateObject = System.DateTime;
+
+namespace Microsoft.Recognizers.Text.DateTime
+{
+    public static class TasksModeProcessing
+    {
+        public const string ParserTypeName = "datetimeV2";
+
+        public static readonly string DateMinString = DateTimeFormatUtil.FormatDate(DateObject.MinValue);
+
+        /*
+        TasksModeModification modifies past datetime references under tasksmode.
+        Eg if input text is 22 june at 9 pm and current time is 22 june 2022, 8 am then
+        under default mode pastdateime value will be 22 june 2022 9 pm, but since time has not been passed
+        under tasksmode it's value will get mapped to 22 june 2021 9 pm.
+        */
+        public static DateTimeParseResult TasksModeModification(DateTimeParseResult slot, DateObject referenceTime)
+        {
+            switch (slot.Type.Substring(ParserTypeName.Length + 1))
+            {
+                case Constants.SYS_DATETIME_DATE:
+                    slot = TasksModeModifyDateValue(slot, referenceTime);
+                    break;
+            }
+
+            return slot;
+        }
+
+        /*
+         Change resolution value of datetime value under tasksmode.
+         */
+        public static TimeOfDayResolutionResult TasksModeResolveTimeOfDay(string tod)
+        {
+            var result = new TimeOfDayResolutionResult();
+            switch (tod)
+            {
+                case Constants.EarlyMorning:
+                    result.Timex = Constants.EarlyMorning;
+                    result.BeginHour = 6;
```

Comment:
Please declare your own constants in this file instead of having only the hardcoded numbers.

Reply:
It makes understanding and maintenance easier.

Reply:
Created a separate constant file for tasksmode constants.